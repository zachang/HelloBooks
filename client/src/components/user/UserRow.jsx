import React from 'react';
import PropTypes from 'react-proptypes';

/**
 * @function UserRow
 * @param {object} props
 * @return {XML} JSX
 */
const UserRow = props => (
  <tr>
    <td>{ props.user.fullname }</td>
    <td>{ props.user.username }</td>
    <td>{ props.user.phoneNo }</td>
    <td>{ props.user.email }</td>
  </tr>
);

UserRow.propTypes = {
  user: PropTypes.object.isRequired,

};
export default UserRow;
