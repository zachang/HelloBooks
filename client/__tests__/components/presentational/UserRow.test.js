import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import UserRow from '../../../src/components/user/UserRow.jsx';

describe('<UserRow/>', () => {
  const props = {
    user: {
      fullname: 'Nana Stack',
      username: 'Nandy09',
      phoneNo: '09077777777',
      email: 'Nandy09@gmail.com'
    }
  };

  const shallowWrapper = shallow(<UserRow {...props}/>);
  const tree = toJson(shallowWrapper);


  it('renders <UserRow/> component', () => {
    expect(tree).toMatchSnapshot();
  });
});

