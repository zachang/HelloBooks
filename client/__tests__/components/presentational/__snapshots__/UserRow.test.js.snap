// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserRow/> renders <UserRow/> component 1`] = `
<tr>
  <td>
    Nana Stack
  </td>
  <td>
    Nandy09
  </td>
  <td>
    09077777777
  </td>
  <td>
    Nandy09@gmail.com
  </td>
</tr>
`;
